README.md
# ReactJS Collapsible component
[![NPM version][npm-image]][npm-url]
[![Downloads][downloads-image]][npm-url]

ReactJs collapsible accordion component. Light, simple and functional. Allows expand and collapse html content. Ready for use in your ReactJs application. Works with React 16.x.x versions.

## Demo

[https://www.devzonetech.com/demo/react-collapsible-component/](https://www.devzonetech.com/demo/react-collapsible-component/)

## Installation

`npm install react-collapsible-component --save`

## Usage

Howto use ReactJs collapsible accordion component.

# 1. Regular usage

```jsx
import {
    CollapsibleComponent,
    CollapsibleHead,
    CollapsibleContent
} from "react-collapsible-component";

export default class MyComponent extends React.Component {
    constructor() {
        super();
    }

    render() {
        return (
            <div>
                ...
                <CollapsibleComponent>
                    <CollapsibleHead className="additionalClassForHead">
                        Head title 1
                    </CollapsibleHead>
                    <CollapsibleContent className="additionalClassForContent">
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis
                            nostrud exercitation ullamco laboris nisi ut aliquip
                            ex{" "}
                        </p>
                    </CollapsibleContent>

                    <CollapsibleHead isExpanded={true}>
                        Head title 2
                    </CollapsibleHead>
                    <CollapsibleContent isExpanded={true}>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis
                            nostrud exercitation ullamco laboris nisi ut aliquip
                            ex ea commodo consequat. Duis aute irure dolor in
                            reprehenderit in voluptate velit esse cillum dolore
                            eu fugiat nulla pariatur. Excepteur sint occaecat
                            cupidatat non proident, sunt in culpa qui officia
                            deserunt mollit anim id est laborum.
                        </p>
                    </CollapsibleContent>

                    <CollapsibleHead>Head title 3</CollapsibleHead>
                    <CollapsibleContent>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis
                            nostrud exercitation ullamco laboris nisi ut aliquip
                            ex ea commodo consequat.{" "}
                        </p>
                    </CollapsibleContent>
                </CollapsibleComponent>
            </div>
        );
    }
}
```

# 2. Use with collapsed content positioned above some of Head titles

```jsx
import {
    CollapsibleComponent,
    CollapsibleHead,
    CollapsibleContent
} from "react-collapsible-component";

export default class MyComponent extends React.Component {
    constructor() {
        super();
    }

    render() {
        return (
            <div>
                ...
                <CollapsibleComponent name="second">
                    <CollapsibleHead>Head title 1</CollapsibleHead>
                    <CollapsibleContent>
                        <p>
                            11111 Lorem ipsum dolor sit amet, consectetur
                            adipiscing elit, sed do eiusmod tempor incididunt ut
                            labore et dolore magna aliqua. Ut enim ad minim
                            veniam, quis nostrud exercitation ullamco laboris
                            nisi ut aliquip ex ea commodo consequat.
                        </p>
                    </CollapsibleContent>

                    <CollapsibleContent isExpanded={true}>
                        <p>
                            22222 Lorem ipsum dolor sit amet, consectetur
                            adipiscing elit, sed do eiusmod tempor incididunt ut
                            labore et dolore magna aliqua. Ut enim ad minim
                            veniam, quis nostrud exercitation ullamco laboris
                            nisi ut aliquip ex ea commodo consequat. Duis aute
                            irure dolor in
                        </p>
                    </CollapsibleContent>
                    <CollapsibleHead
                        showContentAboveButton={true}
                        isExpanded={true}
                    >
                        Head title 2
                    </CollapsibleHead>

                    <CollapsibleContent>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit.
                        </p>
                    </CollapsibleContent>
                    <CollapsibleHead>Head title 3</CollapsibleHead>

                    <CollapsibleHead>Head title 4</CollapsibleHead>
                    <CollapsibleContent>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis
                            nostrud exercitation ullamco laboris nisi ut aliquip
                            ex ea commodo consequat. Duis aute irure dolor in
                            reprehenderit.
                        </p>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis
                            nostrud exercitation ullamco laboris nisi ut aliquip
                            ex ea commodo consequat.
                        </p>
                    </CollapsibleContent>

                    <CollapsibleHead>Head title 5</CollapsibleHead>
                    <CollapsibleContent>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis
                            nostrud exercitation ullamco laboris nisi ut aliquip
                            ex ea commodo consequat.
                        </p>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor.
                        </p>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua.
                        </p>
                    </CollapsibleContent>
                </CollapsibleComponent>
            </div>
        );
    }
}
```

## Component Props

| _Prop_     | _Default_ | _Required_                                     | _Values_               | Description                       |
| ---------- | --------- | ---------------------------------------------- | ---------------------- | --------------------------------- |
| className  |           | no                                             | String                 | Custom defined css class          |
| name       |           | if more than one conponent on one page is used | String                 | Qunique name identifier           |
| isExpanded | false     | no                                             | Boolean: true or false | Show or hide collapsible content. |
| showContentAboveButton | false     | no                                             | Boolean: true or false | Show collapsible content above Head title. |


## License

This project is licensed under the MIT License - see the [LICENSE](LICENSE) file for details

[DEVZONE TECH Ltd.](https://www.devzonetech.com/)

[npm-url]: https://npmjs.org/package/react-collapsible-component
[downloads-image]: http://img.shields.io/npm/dm/react-collapsible-component.svg
[npm-image]: https://badge.fury.io/js/react-collapsible-component.svg


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

## [1.3.4] - 2020-01-31
### Changed
Dependencies updated.

### Fix
-  clicking on <CollapsibleContent> it triggers form submit event #2 

## [1.3.3] - 2019-12-20
### Changed
- Readme file updated.
- Added npm information
- PeerDependencies versioning improvement

## [1.3.0] - 2019-12-13
### Changed
- [#1](https://github.com/devzonetech/react-collapsible-component/issues/1) Putting Head below Content
- Code refactoring
- New Demo application added
- Ability to be used several components on one page
- Package size reduced


The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

demo/src/App.js
import React from "react";
import "./App.css";
import {
    CollapsibleComponent,
    CollapsibleHead,
    CollapsibleContent
} from "react-collapsible-component";

function App() {
    return (
        <div className="App">
            <div id="wrapper">
                <h1>Demo CollapsibleComponent ReactJs component</h1>
                <div>
                    <p>
                        ReactJs collapsible accordion component. Light, simple
                        and functional. Allows expand and collapse html content.
                        Ready for use in your ReactJs application.
                    </p>
                </div>

                <div>
                    <h2>Demo with content bellow Head titles</h2>

                    <CollapsibleComponent name="first">
                        <CollapsibleHead className="additionalClassForHead">
                            Head title 1
                        </CollapsibleHead>
                        <CollapsibleContent className="additionalClassForContent">
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex{" "}
                            </p>
                        </CollapsibleContent>

                        <CollapsibleHead isExpanded={true}>
                            Head title 2
                        </CollapsibleHead>
                        <CollapsibleContent isExpanded={true}>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat. Duis aute irure dolor in
                                reprehenderit in voluptate velit esse cillum
                                dolore eu fugiat nulla pariatur. Excepteur sint
                                occaecat cupidatat non proident, sunt in culpa
                                qui officia deserunt mollit anim id est laborum.
                            </p>
                        </CollapsibleContent>

                        <CollapsibleHead>Head title 3</CollapsibleHead>
                        <CollapsibleContent>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat.{" "}
                            </p>
                        </CollapsibleContent>
                    </CollapsibleComponent>
                </div>

                <div style={{marginTop: '100px'}}>
                    <h2>Demo - content positioned above some of Head titles</h2>
                    <CollapsibleComponent name="second">
                        <CollapsibleHead>Head title 1</CollapsibleHead>
                        <CollapsibleContent>
                            <p>
                                11111 Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat.
                            </p>
                        </CollapsibleContent>

                        <CollapsibleContent isExpanded={true}>
                            <p>
                                22222 Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat. Duis aute irure dolor in
                            </p>
                        </CollapsibleContent>
                        <CollapsibleHead
                            showContentAboveButton={true}
                            isExpanded={true}
                        >
                            Head title 2
                        </CollapsibleHead>

                        <CollapsibleContent>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit.
                            </p>
                        </CollapsibleContent>
                        <CollapsibleHead>Head title 3</CollapsibleHead>

                        <CollapsibleHead>Head title 4</CollapsibleHead>
                        <CollapsibleContent>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat. Duis aute irure dolor in
                                reprehenderit.
                            </p>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat.
                            </p>
                        </CollapsibleContent>

                        <CollapsibleHead>Head title 5</CollapsibleHead>
                        <CollapsibleContent>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua. Ut
                                enim ad minim veniam, quis nostrud exercitation
                                ullamco laboris nisi ut aliquip ex ea commodo
                                consequat.
                            </p>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor.
                            </p>
                            <p>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit, sed do eiusmod tempor
                                incididunt ut labore et dolore magna aliqua.
                            </p>
                        </CollapsibleContent>
                    </CollapsibleComponent>
                </div>
            </div>
        </div>
    );
}

export default App;
